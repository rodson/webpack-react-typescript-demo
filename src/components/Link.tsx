import React, { ReactNode } from "react";
const Link = ({ page }: { page: string, children?: ReactNode }) => {
  return (
    <a
      className="normal"
      href={page}
      onMouseEnter={() => console.log("Mouse enter")}
      onMouseLeave={() => console.log("Mouse leave")}
    >
      My Domain
    </a>
  );
};
export default Link;