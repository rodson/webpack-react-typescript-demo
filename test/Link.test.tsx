import { cleanup, render } from "@testing-library/react";
import Link from '../src/components/Link';

afterEach(() => {
  cleanup()
});

it('renders correctly', () => {
  const { asFragment }  = render(<Link page='http://www.tencent.com'>Bonarhyme</Link>)
  expect(asFragment()).toMatchSnapshot();
});