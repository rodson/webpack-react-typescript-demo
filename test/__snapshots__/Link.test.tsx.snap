// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <a
    class="normal"
    href="http://www.tencent.com"
  >
    My Domain
  </a>
</DocumentFragment>
`;
